### ISSUE TITLE
Create manuscript repo without rootstock commit history

### ISSUE THREAD

**Issue by eric-czech** [manubot/rootstock#398]
Hey @dhimmel, OOC I tried creating a [repo](https://github.com/pystatgen/sgkit-publication) fresh instead of from a clone as suggested in https://github.com/manubot/rootstock/blob/master/SETUP.md.  I did that to avoid bringing in the commit history from the rootstock repo.  Is there a way to make this work?  The first problem I ran into is in [get_rootstock_commit](https://github.com/manubot/manubot/blob/0c2e5b28809acd7c9ca76152146ddc24345d0781/manubot/process/metadata.py#L181) when running `build.sh`. 

Are there other ways you would recommend to keep the paper repo up-to-date with the template repo?

**Comment by dhimmel** [comment]
Hi Eric! We've played around with a squashing all rootstock commits upon creating a manuscript in https://github.com/manubot/rootstock/pull/65. The main problem it becomes much harder to pull rootstock updates in the future.

> avoid bringing in the commit history from the rootstock repo

I agree it's undesirable to clone the entire rootstock commit history. In the long-term, we hope to switch to using a GitHub template repo that would be located at https://github.com/manubot/template. But for now we don't have a great solution to creating a manuscript without the rootstock commit history.

> The first problem I ran into is in get_rootstock_commit when running build.sh

Yes, `get_rootstock_commit` will fail without the entire rootstock commit history, which will cause a warning in the logs and prevent a metadata value from getting set. `get_rootstock_commit` might also fail if you've switched the default branch name from `master` to `main` (see https://github.com/manubot/rootstock/issues/349). But despite being annoying in the logs, this warning can be ignored.

In summary, I'd caution against squashing everything to start unless you're willing to do extra maintenance work down the road. But if you'd like to squash, happy to help you debut any further issues.

> The first problem I ran into is in get_rootstock_commit when running build.sh

Any subsequent issues?

**Comment by eric-czech** [comment]
> We've played around with a squashing all rootstock commits upon creating a manuscript in #65. The main problem is that it becomes much harder to pull rootstock updates in the future.

Ok, I'll use as intended then.

> Any subsequent issues?

Nope!  Had to replace master with main everywhere in manubot.yaml (so far only that), but the `build.sh` scripts runs without errors after rebasing on the latest rootstock commit.

**Comment by dhimmel** [comment]
Nice.  This issue is in relation to https://github.com/pystatgen/sgkit-publication right?

**Comment by eric-czech** [comment]
> This issue is in relation to https://github.com/pystatgen/sgkit-publication right?

Yep.
